tests/rptest/tests/redpanda_kerberos_test.py

```diff
@@ -97,29 +102,78 @@ def setUp(self):
         for node in self.redpanda.nodes:
             self._configure_service_node("redpanda", node)
 
-        for node in self.client.nodes:
-            self._configure_client_node("client", node)
+        self._configure_client_node("client")
 
 
 class RedpandaKerberosTest(RedpandaKerberosTestBase):
     def __init__(self, test_context, **kwargs):
         super(RedpandaKerberosTest, self).__init__(test_context, **kwargs)
 
     @cluster(num_nodes=5)
-    def test_init(self):
+    @parametrize(req_principal="client",
+                 acl=True,
+                 topics=["needs_acl", "always_visible"],
+                 fail=False)
+    @parametrize(req_principal="client",
+                 acl=False,
+                 topics=["always_visible"],
+                 fail=False)
+    @parametrize(req_principal="invalid", acl=False, topics={}, fail=True)
+    def test_init(self, req_principal: str, acl: bool, topics: set[str],
+                  fail: bool):
+        def is_auth_error(msg):
+            return b'No Kerberos credentials available' in msg
+
         feature_name = "kafka_gssapi"
         self.redpanda.logger.info(f"Principals: {self.kdc.list_principals()}")
         admin = Admin(self.redpanda)
         admin.put_feature(feature_name, {"state": "active"})
         self.redpanda.await_feature_active(feature_name, timeout_sec=30)
 
-        kcat = KafkaCat(self.redpanda)
-        metadata = kcat.metadata()
-        self.redpanda.logger.info(f"Metadata (SCRAM): {metadata}")
-        assert len(metadata['brokers']) == 3
-        metadata = self.client.metadata()
-        self.redpanda.logger.info(f"Metadata (GSSAPI): {metadata}")
-        assert len(metadata['brokers']) == 3
+        username, password, mechanism = self.redpanda.SUPERUSER_CREDENTIALS
+        super_rpk = RpkTool(self.redpanda,
+                            username=username,
+                            password=password,
+                            sasl_mechanism=mechanism)
+
+        # Create a topic that's visible to "client" iff acl = True
+        client_user_principal = f"User:client"
```

Comment:
This is actually a "happy path" test for using the rules.  The only rule, by default is `DEFAULT`.  If the kerberos principal's realm matches the default realm (`@EXAMPLE.COM`) the returned mapped name will just be the primary, which in this case is `client`.  So `client@EXAMPLE.COM` --> `client`.

Reply:
Nice!

Reply:
Nice.